fineract-core/src/main/java/org/apache/fineract/portfolio/calendar/data/CalendarData.java

```diff
@@ -194,6 +194,51 @@ private CalendarData(LocalDate startDate, boolean repeating, EnumOptionData freq
         this.duration = null;
     }
 
+    public CalendarData(String title, String description, LocalDate startDate, boolean repeating, EnumOptionData frequency,
+            Integer interval, EnumOptionData repeatsOnDay, String dateFormat, String locale, String typeId) {
+        this.title = title;
+        this.description = description;
+        this.startDate = startDate;
+        this.repeating = repeating;
+        this.frequency = frequency;
+        this.interval = interval;
+        this.repeatsOnDay = repeatsOnDay;
+        this.dateFormat = dateFormat;
+        this.locale = locale;
+        this.typeId = typeId;
+
+        this.id = null;
```

Comment:
Neither of the fields are final, so you dont need to set `null` to them... kindly remove all unnecessary field initialisation